url56b1dc76d3_wufunxiang_Libs_tgz-pJ8-org_msgpack_util_TemplatePrecompilerJ8/src/org/msgpack/template/builder/beans/StandardBeanInfo.java

```diff
@@ -0,0 +1,1505 @@
+        HashMap<String, EventSetDescriptor> map = new HashMap<String, EventSetDescriptor>();
+        for (int i = 0; i < theDescs.length; i++) {
+            map.put(theDescs[i].getName(), theDescs[i]);
+        }
+        return map;
+    }
+
+    private static String getQualifiedName(Method method) {
+        String qualifiedName = method.getName();
+        Class<?>[] paramTypes = method.getParameterTypes();
+        if (paramTypes != null) {
+            for (int i = 0; i < paramTypes.length; i++) {
+                qualifiedName += "_" + paramTypes[i].getName(); //$NON-NLS-1$
+            }
+        }
+        return qualifiedName;
+    }
+
+    /**
+     * Introspects the supplied class and returns a list of the public methods
+     * of the class
+     * 
+     * @return An array of MethodDescriptors with the public methods. null if
+     *         there are no public methods
+     */
+    private MethodDescriptor[] introspectMethods() {
+        return introspectMethods(false, beanClass);
+    }
+
+    private MethodDescriptor[] introspectMethods(boolean includeSuper) {
+        return introspectMethods(includeSuper, beanClass);
+    }
+
+    private MethodDescriptor[] introspectMethods(boolean includeSuper,
+            Class<?> introspectorClass) {
+
+        // Get the list of methods belonging to this class
+        Method[] basicMethods = includeSuper ? introspectorClass.getMethods()
+                : introspectorClass.getDeclaredMethods();
+
+        if (basicMethods == null || basicMethods.length == 0)
+            return null;
+
+        ArrayList<MethodDescriptor> methodList = new ArrayList<MethodDescriptor>(
+                basicMethods.length);
+
+        // Loop over the methods found, looking for public non-static methods
+        for (int i = 0; i < basicMethods.length; i++) {
+            int modifiers = basicMethods[i].getModifiers();
+            if (Modifier.isPublic(modifiers)) {
+                // Allocate a MethodDescriptor for this method
+                MethodDescriptor theDescriptor = new MethodDescriptor(
+                        basicMethods[i]);
+                methodList.add(theDescriptor);
+            }
+        }
+
+        // Get the list of public methods into the returned array
+        int methodCount = methodList.size();
+        MethodDescriptor[] theMethods = null;
+        if (methodCount > 0) {
+            theMethods = new MethodDescriptor[methodCount];
+            theMethods = methodList.toArray(theMethods);
+        }
+
+        return theMethods;
+    }
+
+    /**
+     * Introspects the supplied class and returns a list of the Properties of
+     * the class
+     * 
+     * @param stopClass -
+     *            the to introspecting at
+     * @return The list of Properties as an array of PropertyDescriptors
+     * @throws IntrospectionException
+     */
+    @SuppressWarnings("rawtypes")
+	private PropertyDescriptor[] introspectProperties(Class<?> stopClass)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 21. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 675-680 into a separate method.